crates/bevy_diagnostic/src/diagnostic.rs

```diff
@@ -200,29 +282,29 @@ impl Diagnostic {
 pub struct DiagnosticsStore {
     // This uses a [`StableHashMap`] to ensure that the iteration order is deterministic between
     // runs when all diagnostics are inserted in the same order.
-    diagnostics: StableHashMap<DiagnosticId, Diagnostic>,
+    diagnostics: StableHashMap<DiagnosticPath, Diagnostic>,
```

Comment:
This double hashes the hash, which increases the odds of collision. Additionally, even single hashes aren't perfect and produce collisions, which is why Rust's HashMap impl assumes a fall back to Eq/PartialEq in the event of collision. However the Eq/PartialEq _also_ uses the hash, which makes this a problem.

 See the PreHashMap in bevy_util for one implementation of "pre hashing without double hashing and proper PartialEq/Eq fallback".